### PR TITLE
NetCore: Replaces the usages of HttpContext.Items ScopeProvider

### PR DESCRIPTION
**Pull request by bergmania**
Introduces a new interface `IRequestCache` and uses that in `ScopeProvider` instead of `HttpContext.Current.Items`

The interface is implemented by our `HttpRequestAppCache`. We could consider creating a new class that just implements the needed methods.

### CODE REVIEW

File: src/Umbraco.Core/Scoping/ScopeProvider.cs [hunk 1]
```diff
@@ -181,51 +182,31 @@ private static void SetCallContextObject(string key, IInstanceIdentifiable value
...
+        private bool SetHttpContextObject(string key, object value, bool required = true)
         {
-            var httpContextItems = HttpContextItems;
-            if (httpContextItems == null)
-            {
-                if (required)
-                    throw new Exception("HttpContext.Current is null.");
-                return false;
-            }
-#if DEBUG_SCOPES
```
Comment (Shazwazza) [inline src/Umbraco.Core/Scoping/ScopeProvider.cs]: did you want to remove this directive? I've never used it but never had to debug scopes before

File: src/Umbraco.Tests/Scoping/ScopeTests.cs [hunk 2]
```diff
@@ -442,7 +429,7 @@ public void CallContextScope2()
             Assert.IsNull(scopeProvider.AmbientScope);
 
             var httpContextItems = new Hashtable();
-            ScopeProviderStatic.HttpContextItemsGetter = () => httpContextItems;
+        //    ScopeProviderStatic.HttpContextItemsGetter = () => httpContextItems;
```
Comment (Shazwazza) [inline src/Umbraco.Tests/Scoping/ScopeTests.cs]: Can prob remove these commented out lines

File: src/Umbraco.Core/Scoping/ScopeProvider.cs (hunk 1 above)
Comment (bergmania) [reply src/Umbraco.Core/Scoping/ScopeProvider.cs]: I'm a bit unsure how to port it.. Do you have any ideas?

Comment (Shazwazza) [reply src/Umbraco.Core/Scoping/ScopeProvider.cs]: @bergmania I've pushed an update to this branch, see rev 898167f8c60f7ac093d4fe12205b5adf9506619f

Let me know what you think.

**Comment by bergmania** [comment]
I like the changes.. I figure out why this leads to failing tests

**Pull request merged**